synapse/app/_base.py

```diff
@@ -129,17 +128,8 @@ def run():
             if print_pidfile:
                 print(pid_file)
 
-            daemon = Daemonize(
-                app=appname,
-                pid=pid_file,
-                action=run,
-                auto_close_fds=False,
-                verbose=True,
-                logger=logger,
-            )
-            daemon.start()
-        else:
-            run()
+            daemonize_process(pid_file, logger)
+        run()
```

Comment:
I've realised that the original code wrapped this with a `try/except` for a reason, which is that after stderr gets redirected to /dev/null, any uncaught exceptions will otherwise get thrown away.

I'm currently wondering how best to solve that.